lock-api-objects/src/main/java/com/palantir/lock/watch/CommitUpdate.java

```diff
@@ -68,15 +79,22 @@ interface InvalidateSome extends CommitUpdate {
 
         Set<LockDescriptor> invalidatedLocks();
 
+        Map<LockDescriptor, List<ChangeMetadata>> aggregatedMetadata();
+
         @Override
         default <T> T accept(Visitor<T> visitor) {
-            return visitor.invalidateSome(invalidatedLocks());
+            return visitor.invalidateSome(invalidatedLocks(), aggregatedMetadata());
         }
     }
 
     interface Visitor<T> {
         T invalidateAll();
 
         T invalidateSome(Set<LockDescriptor> invalidatedLocks);
+
+        default T invalidateSome(
+                Set<LockDescriptor> invalidatedLocks, Map<LockDescriptor, List<ChangeMetadata>> aggregatedMetadata) {
+            return invalidateSome(invalidatedLocks);
+        }
```

Comment:
By using a `default` method here, we avoid a break on the Java layer, but still allow custom visitors to override the behavior. This also avoids having to refactor a bunch of tests. However, I think it's a bit ugly and not that maintainable. The alternative is to change the visitor interface method to always include aggregated metadata. This requires Atlas users to update their commit update visitors when they pull in the new version.

Reply:
Yeah we want to break the API here